lib/CoreBot.js

```diff
@@ -610,31 +610,48 @@ function Botkit(configuration) {
     };
 
     botkit.hears = function(keywords, events, cb) {
-        if (typeof(keywords) == 'string') {
-            keywords = [keywords];
-        }
-        if (typeof(events) == 'string') {
-            events = events.split(/\,/g);
+        var matchFn;
+        var theRule;
+
+        // Compile
+        if (typeof(keywords) == 'function') {
```

Comment:
both `==` and `===` works. It is a string comparison and  `==` was in the previous code.